### PR TITLE
add create-digest built-in function

### PR DESCRIPTION
**Pull request by wusphinx**

### CODE REVIEW

**Comment by shenli** [comment]
@wusphinx Thanks for your PR! Please sign the CLA.

**Comment by wusphinx** [comment]
@shenli Sorry about that, I have signed

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -364,7 +364,50 @@ type builtinCreateDigestSig struct {
...
+		return d, errors.Trace(err)
+	}
+
+	bin, err := args[1].ToBytes()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	var hasher hash.Hash
+	switch digestType {
```
Comment (shenli) [inline expression/builtin_encryption.go]: digestType = strings.ToUpper(digestType)

File: expression/builtin_encryption.go [hunk 2]
```diff
@@ -364,7 +364,50 @@ type builtinCreateDigestSig struct {
...
+	case "SHA224":
```
Comment (shenli) [inline expression/builtin_encryption.go]: Make these digest type names as consts.

File: expression/builtin_encryption.go [hunk 3]
```diff
@@ -364,7 +364,50 @@ type builtinCreateDigestSig struct {
...
+		hasher = sha256.New224()
+	case "SHA256":
+		hasher = sha256.New()
+	case "SHA384":
+		hasher = sha512.New384()
+	case "SHA512":
+		hasher = sha512.New()
+	}
+
+	if hasher != nil {
```
Comment (shenli) [inline expression/builtin_encryption.go]: if hasher == nil {
    return
}

File: expression/builtin_encryption.go (hunk 1 above)
Comment (wusphinx) [inline expression/builtin_encryption.go]: update for that

File: expression/builtin_encryption.go [hunk 4]
```diff
@@ -362,9 +363,61 @@ type builtinCreateDigestSig struct {
 	baseBuiltinFunc
 }
 
+// Supported digest type name
+const (
+	SHA224NAME string = "SHA224"
```
Comment (shenli) [inline expression/builtin_encryption.go]: The Golang standard libs use MixedCaps for const name. So it is better to rename it to SHA224Name.

Comment (wusphinx) [inline expression/builtin_encryption.go]: updated for that

**Comment by tiancaiamao** [comment]
This is a **MySQL Enterprise** function, I run it in a MySQL 5.6 Community Server:

```
mysql> select create_digest('sha224', 'hello world');
ERROR 1305 (42000): FUNCTION create_digest does not exist
```

@shenli

**Comment by coocood** [comment]
@wusphinx 
I'm sorry, we should not have added the syntax of this enterprise function in the first place.
Do you have an enterprise edition of MySQL when you implement this function?

**Comment by wusphinx** [comment]
@coocood actually,I have not an enterprise edition of MySQL, I wrote the function according to the document
so, this function will be cancelled?

**Comment by coocood** [comment]
@wusphinx 
The documentation usually does not cover all the corner cases, we can not make sure the implementation is correct without comparing to the result of a running MySQL server.

So I'm sorry, we have to close it.

**Pull request closed without merging**